tools/openapi-generator/config/go-templates/cmd.mustache

```diff
@@ -123,30 +140,79 @@ func (opts *{{#lambda.camelcase}}{{operationId}}{{/lambda.camelcase}}Opts) run(c
 		{{/bodyParam}}
 	}
 
-	{{#returnType}}resp, {{/returnType}}_, err := opts.client.{{classname}}.{{operationId}}WithParams(ctx, params).Execute()
-	{{#returnType}}
-	if err != nil {
-		return err
+	var err error
+	{{#returnType}}opts.resp, {{/returnType}}_, err = opts.client.{{classname}}.{{operationId}}WithParams(ctx, params).Execute()
+	return err
+}
+
+{{#vendorExtensions.x-watcher-operation}}
+func (opts *{{#lambda.camelcase}}{{operationId}}{{/lambda.camelcase}}Opts) watch(ctx context.Context) error {
+	if !opts.watchEnabled {
+		return nil
 	}
 
-	prettyJSON, errJson := json.MarshalIndent(resp, "", " ")
+	spin := spinner.New(spinner.CharSets[9], 100 * time.Millisecond)
+	spin.Start()
+	defer spin.Stop()
+	
+	watchCtx, cancel := context.WithTimeout(ctx, opts.wt)
+	defer cancel()
+	for {
+		select {
+			case <-watchCtx.Done():
+				return watchCtx.Err()
+			default:
+				params := &admin.{{#lambda.titlecase}}{{.}}{{/lambda.titlecase}}ApiParams{
+					{{#vendorExtensions.x-watcher-params.0}}
+					{{key}}: *opts.resp.{{value}},
+					{{/vendorExtensions.x-watcher-params.0}}
+					{{#vendorExtensions.x-watcher-params.1}}
+					{{key}}: *opts.resp.{{value}},
+					{{/vendorExtensions.x-watcher-params.1}}
+					{{#vendorExtensions.x-watcher-params.2}}
+					{{key}}: *opts.resp.{{value}},
+					{{/vendorExtensions.x-watcher-params.2}}
+				}
+				resp, _, err := opts.client.{{classname}}.{{#lambda.titlecase}}{{.}}{{/lambda.titlecase}}WithParams(ctx, params).Execute()
+				if err != nil {
+					return err
+				}
+				if resp.{{vendorExtensions.x-watcher-return-property}} != nil && *resp.{{vendorExtensions.x-watcher-return-property}} == "{{vendorExtensions.x-watcher-state}}" {
+					return nil
+				}
```

Comment:
probably a `time.Sleep(2 * time.Second)` would help here to avoid rate limiting

Reply:
done! added 4 since that is the default for the current watcher